server/common/src/test/java/org/eclipse/lsp/cobol/common/mapping/ExtendedTextTest.java

```diff
@@ -373,4 +375,304 @@ void testInsertWithPadding() {
     assertEquals(
         expectedCopybook.getLines().get(2).toString(), copybook1.getLines().get(2).toString());
   }
+
+  @Test
+  void testReplace_three_lines() {
+    ExtendedText extendedText = new ExtendedText("MOVE VAR1 TO VAR2", "uri");
+
+    Range range = new Range(new Position(0, 0), new Position(0, 18));
```

Comment:
The end position seems to be outside of the extendedText valid range, yet no exception is produced.